lib/posix/pthread.c

```diff
@@ -160,6 +152,20 @@ int pthread_create(pthread_t *newthread, const pthread_attr_t *attr,
 	if (pthread_num >= CONFIG_MAX_PTHREAD_COUNT) {
 		return EAGAIN;
 	}
+	if ((attr == NULL) || (attr->initialized == 0U)
```

Comment:
These double-parens look really weird, please don't inherit them from previous code you change. But bigger problem is that this check doesn't look correct. It could be such only if you take frivolous interpretation of this phrase from http://pubs.opengroup.org/onlinepubs/9699919799/functions/pthread_create.html:

> The behavior is undefined if the value specified by the attr argument to pthread_create() does not refer to an initialized thread attributes object.

So, if such behavior is undefined, you *could* treat uninitialized attrs as NULL attrs. But this way, we only complicate our own life with handling more attrs, etc.

Please add `attr != NULL && attr->initialized == 0` check where previous EINVAL return was, this should simplify and clarify behavior for now and future.

Reply:
You are right, it is better not to do anything when the behavior is undefined. This also allowed me to restore the test I originally removed.